fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RecordKeyCursorTest.java

```diff
@@ -0,0 +1,270 @@
+import com.apple.foundationdb.record.RecordCursorContinuation;
+import com.apple.foundationdb.record.RecordCursorResult;
+import com.apple.foundationdb.record.ScanProperties;
+import com.apple.foundationdb.record.TestRecords1Proto;
+import com.apple.foundationdb.record.TupleRange;
+import com.apple.foundationdb.tuple.Tuple;
+import com.google.common.base.Strings;
+import com.google.protobuf.Message;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class RecordKeyCursorTest extends FDBRecordStoreTestBase {
+    public enum UseContinuations { NONE, CONTINUATIONS, BYTE_LIMIT }
+
+    public static Stream<Arguments> splitContinuationVersion() {
+        return Stream.of(true, false)
+                .flatMap(
+                        split -> Arrays.stream(UseContinuations.values())
+                                .flatMap(useContinuations -> Stream.of(
+                                                FDBRecordStore.RECORD_COUNT_KEY_ADDED_FORMAT_VERSION, // 3
+                                                FDBRecordStore.SAVE_VERSION_WITH_RECORD_FORMAT_VERSION, // 6
+                                                FDBRecordStore.MAX_SUPPORTED_FORMAT_VERSION)
+                                        .map(version -> Arguments.of(split, useContinuations, version))));
+    }
+
+    @ParameterizedTest(name = "testIterateRecordsNoIssue [splitLongRecords = {0}, useContinuations = {1}, formatVersion = {2}]")
+    @MethodSource("splitContinuationVersion")
+    void testIterateRecordsNoIssue(boolean splitLongRecords, UseContinuations useContinuations, int formatVersion) throws Exception {
+        final RecordMetaDataHook hook = getRecordMetaDataHook(splitLongRecords);
+        saveRecords(splitLongRecords, formatVersion, hook);
+        // Scan records
+        ScanProperties scanProperties = getScanProperties(useContinuations);
+        final List<Tuple> actualKeys;
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            actualKeys = scanKeys(store, scanProperties, useContinuations != UseContinuations.NONE);
+            commit(context);
+        }
+        List<Tuple> expectedKeys = IntStream.range(1, 101).boxed().map(Tuple::from).collect(Collectors.toList());
+        assertEquals(expectedKeys, actualKeys);
+    }
+
+    @ParameterizedTest(name = "testIterateRecordsMissingRecord [splitLongRecords = {0}], useContinuations = {1}")
```

Comment:
I don't think the names are necessary, but if you're going to have them, they should be consistent.
```suggestion
    @ParameterizedTest(name = "testIterateRecordsMissingRecord [splitLongRecords = {0}, useContinuations = {1}, formatVersion = {2}]")
```

Reply:
Done.